### ISSUE TITLE
MD014: `rumdl` only finds first violation in a code block

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#516]
### rumdl version

0.1.45

### Installation method

mise

### Bug description

Rule MD014 (https://github.com/rvben/rumdl/blob/a304d288c836b694e792d329a782db3e3165858c/docs/md014.md) ensures that commands in fenced code blocks that are prefixed with a prompt have output.

This works well for the first instance of a command in a code block, but not subsequent instances.

### Steps to reproduce

1. Create a Markdown file with this content:
   ````markdown
   ---
   title: Heading
   ---
   
   Here is a fenced code block:
   
   ```shell
   # First invocation of my_command
   $ my_command
   
   # Second invocation of my_command
   $ my_command
   ```
   ````
1. Run `rumdl check` on the Markdown file with no specific configuration.

### Expected behavior

Given the scenario above, I'd expect two lines (9 and 12) to be flagged.

### Actual behavior

Given the scenario above, only one line (9) is flagged. If that line is fixed as follows:

````markdown
---
title: Heading
---

Here is a fenced code block:

```shell
# First invocation of my_command
my_command

# Second invocation of my_command
$ my_command
```
````

Then no violations are flagged.

### Configuration

None.

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.1.46. MD014 now reports all `$`-prefixed command lines per code block that have output, not just the first. Thanks for reporting @eread!

**Comment by eread** [comment]
> Fixed in v0.1.46. MD014 now reports all `$`-prefixed command lines per code block that have output, not just the first. Thanks for reporting [@eread](https://github.com/eread)!

Thanks @rvben. I can see that it does now flag all the lines in the scenario. However, it seems as though if the first command line passes, no other lines are checked in the code block.

So this still reports no violations:

````markdown
---
title: Heading
---

Here is a fenced code block:

```shell
# First invocation of my_command
my_command

# Second invocation of my_command
$ my_command
```
````
